--- origsrc/xpdf-4.01/doc/sample-xpdfrc	2019-02-18 13:24:19.000000000 -0500
+++ src/xpdf-4.01/doc/sample-xpdfrc	2019-02-25 15:40:19.565550400 -0500
@@ -4,7 +4,7 @@
 #
 # The Xpdf tools look for a config file in two places:
 # 1. ~/.xpdfrc
-# 2. in a system-wide directory, typically /usr/local/etc/xpdfrc
+# 2. in a system-wide directory, typically /etc/xpdfrc
 #
 # This sample config file demonstrates some of the more common
 # configuration options.  Everything here is commented out.  You
@@ -29,20 +29,20 @@
 # installed in a "standard" location, xpdf will find them
 # automatically.)
 
-#fontFile Times-Roman		/usr/local/share/ghostscript/fonts/n021003l.pfb
-#fontFile Times-Italic		/usr/local/share/ghostscript/fonts/n021023l.pfb
-#fontFile Times-Bold		/usr/local/share/ghostscript/fonts/n021004l.pfb
-#fontFile Times-BoldItalic	/usr/local/share/ghostscript/fonts/n021024l.pfb
-#fontFile Helvetica		/usr/local/share/ghostscript/fonts/n019003l.pfb
-#fontFile Helvetica-Oblique	/usr/local/share/ghostscript/fonts/n019023l.pfb
-#fontFile Helvetica-Bold		/usr/local/share/ghostscript/fonts/n019004l.pfb
-#fontFile Helvetica-BoldOblique	/usr/local/share/ghostscript/fonts/n019024l.pfb
-#fontFile Courier		/usr/local/share/ghostscript/fonts/n022003l.pfb
-#fontFile Courier-Oblique	/usr/local/share/ghostscript/fonts/n022023l.pfb
-#fontFile Courier-Bold		/usr/local/share/ghostscript/fonts/n022004l.pfb
-#fontFile Courier-BoldOblique	/usr/local/share/ghostscript/fonts/n022024l.pfb
-#fontFile Symbol			/usr/local/share/ghostscript/fonts/s050000l.pfb
-#fontFile ZapfDingbats		/usr/local/share/ghostscript/fonts/d050000l.pfb
+#fontFile Times-Roman		/usr/share/fonts/urw-base35/NimbusRoman-Regular.t1
+#fontFile Times-Italic		/usr/share/fonts/urw-base35/NimbusRoman-Italic.t1
+#fontFile Times-Bold		/usr/share/fonts/urw-base35/NimbusRoman-Bold.t1
+#fontFile Times-BoldItalic	/usr/share/fonts/urw-base35/NimbusRoman-BoldItalic.t1
+#fontFile Helvetica		/usr/share/fonts/urw-base35/NimbusSans-Regular.t1
+#fontFile Helvetica-Oblique	/usr/share/fonts/urw-base35/NimbusSans-Italic.t1
+#fontFile Helvetica-Bold	/usr/share/fonts/urw-base35/NimbusSans-Bold.t1
+#fontFile Helvetica-BoldOblique	/usr/share/fonts/urw-base35/NimbusSans-BoldItalic.t1
+#fontFile Courier		/usr/share/fonts/urw-base35/NimbusMonoPS-Regular.t1
+#fontFile Courier-Oblique	/usr/share/fonts/urw-base35/NimbusMonoPS-Italic.t1
+#fontFile Courier-Bold		/usr/share/fonts/urw-base35/NimbusMonoPS-Bold.t1
+#fontFile Courier-BoldOblique	/usr/share/fonts/urw-base35/NimbusMonoPS-BoldItalic.t1
+#fontFile Symbol		/usr/share/fonts/urw-base35/StandardSymbolsPS.t1
+#fontFile ZapfDingbats		/usr/share/fonts/urw-base35/D050000L.t1
 
 # If you need to display PDF files that refer to non-embedded fonts,
 # you should add one or more fontDir options to point to the
@@ -50,7 +50,8 @@
 # .pfb, .ttf, and .ttc files in those directories (other files will
 # simply be ignored).
 
-#fontDir		/usr/local/fonts/bakoma
+#fontDir		/usr/local/share/fonts
+#fontDir		/usr/share/fonts
 
 #----- PostScript output control
 
@@ -58,7 +59,7 @@
 # A4, or A3.  You can also specify a paper size as width and height
 # (in points).
 
-#psPaperSize		letter
+#psPaperSize		A4
 
 #----- text output control
 
@@ -71,7 +72,7 @@
 # Choose the end-of-line convention for multi-line copy-and-past and
 # for pdftotext output.  The available options are unix, mac, and dos.
 
-#textEOL		unix
+textEOL			 unix
 
 #----- misc settings
 
